### PR TITLE
feat: add Starlink probability predictor for future flights

### PR DESCRIPTION
**Pull request by martinamps**
## Problem

Our schedule data only covers **~2 days** (FR24 limitation), but users ask about flights 1-2 weeks out. From an actual MCP conversation:

> "The tool's schedule data only extends about 2 days out, so March 11–12 flights aren't showing yet."

We have **12,000+ historical observations** in `starlink_verification_log` — every time we checked a flight against United.com, we recorded the result. That's a training set.

## Model

Hierarchical Bayesian: time-decayed flight history smoothed toward fleet prior.

**Key insight**: the verification_log is **bimodal** — 77.8% Starlink for tails in `starlink_planes` (verifier bias), 0.1% for random fleet (discovery). Averaging these gives wildly wrong priors. So we use **two priors**:

- **Smoothing prior** (log-conditional, ~77%/0.4%) — for flights WITH history, smooth their empirical rate toward the rate of similar logged flights
- **Cold-start prior** (fleet install rate, ~49%/2%) — for flights WITH NO history, fall back to actual fleet stats from `meta` table

## Backtest results

Holdout: last 48h of prod `starlink_verification_log` (n=210), train on prior 11,634 observations.

| Metric | Value | Baseline |
|--------|-------|----------|
| Accuracy @ 0.5 | **82.4%** | 67.1% (majority class) |
| Brier score | **0.134** | 0.25 (chance) |
| Log loss | 0.469 | — |

**Calibration** (predicted vs actual rate):

| Bucket | n | Predicted | Actual | |
|--------|---|-----------|--------|---|
| [0.3,0.5) | 16 | 0.484 | 0.500 | ✓ |
| [0.5,0.7) | 17 | 0.646 | 0.647 | ✓ |
| [0.7,0.9) | 129 | 0.810 | 0.868 | ✓ |

**Cross-validation**: stable 80-84% across 24h–168h holdout windows. The 168h (7-day) test is actually best (84.4% acc, Brier 0.111) — exactly the forecast horizon we need.

### New tool: `predict_flight_starlink`

```
UA3561 (EWR→JAX): 85% Likely — 4 historical observations, confidence: medium
UA544 (mainline): 0.4% Very unlikely — 1 observation
```

### `check_flight` now falls back to prediction

When no firm schedule data exists:
```
No confirmed aircraft assignment for UA3561 on 2026-03-20 in our database 
(we only track ~2 days of firm schedules).

Probability estimate: ~85% chance of Starlink based on 4 historical 
observation(s) of this flight number (confidence: medium). Check again 
1-2 days before departure for a firm answer.
```

### Performance

Model is cached for 1hr (matches scrape cadence). First call loads 12k rows (~50ms), subsequent calls <1ms.

## Explored & rejected

Used subagents to explore alternatives in parallel:
- **Tail-pool feature** (count distinct tails, not observations): 82.9% acc but **worse Brier (0.171)** — less calibrated. V1 wins 10/15 disagreement cases.
- **Route-level fallback**: only 33% of logged flights have derivable routes — insufficient coverage.
- Simple time-weighted history beats complexity here.

## Verification

```bash
bun run predict-backtest                        # Run backtest on local DB
bun run src/scripts/starlink-predictor.ts --cv  # Cross-validation
bun run src/scripts/starlink-predictor.ts --predict=UA4680
```

### CODE REVIEW

**Comment by martinamps** [comment]
## Update: Deep analysis from parallel subagents incorporated

Ran two background subagents to cross-validate and explore features. Key findings that changed the model:

### 🔴 Time-decay was harmful — removed

Agent found UA3419 (5 obs, all non-Starlink) predicted **50%** because time-decay washed out evidence and the 77% prior took over. Now correctly predicts **22%**.

| Before | After |
|--------|-------|
| UA3419 (5 obs all neg): 50% | **22%** |
| UA3540 (6 obs all neg): 33% | **19%** |
| UA4680 (11 obs all pos): 90% | **96%** |

Empirical backtest confirmed: all-history rate beats time-weighted (Brier 0.102 vs 0.126).

### 🟡 The 82% accuracy is inflated — honest messaging now

Agent ran inverse-propensity reweighting to simulate production distribution (random flights, not our biased verification log). Accuracy drops to ~53%. The test set is drawn from the same biased source as training.

**What IS accurate**: high-confidence predictions (5+ observations, consistent pattern) — these are ~85%+ reliable. MCP output now clearly distinguishes:
- **High** (5+ obs): reliable, strong signal
- **Medium** (2-4 obs): ±20pp uncertainty
- **Low** (0-1 obs): rough prior only, clearly labeled as "treat as upper bound"

### 🟢 Features that DIDN'T help (explored & rejected)

- **Tail-pool** (count distinct tails): slightly higher acc but worse Brier on raw eval
- **Route-level fallback**: only 33% coverage, not worth the complexity
- **Day-of-week**: pure artifact of when verification jobs run (95% Tue/Wed)

### 🟢 Features worth future exploration

- **Operator penalty**: United mainline swaps aircraft ~47% of time even when Starlink tail scheduled. Express carriers hold rotations tighter. Could add ~0.7x multiplier for mainline.
- **Unbiased data collection**: biggest win would be a job that randomly samples *flight numbers* (not tails) to build an unbiased training set over weeks.
